nextcord/ext/commands/bot.py

```diff
@@ -901,6 +903,132 @@ def reload_extension(self, name: str, *, package: Optional[str] = None) -> None:
+            extras: Optional[List[Dict[str, Any]]] = None
+    ) -> List[str]:
+        """Loads all extensions provided in a list.
+        
+        Parameters
+        ----------
+        names: :type:`List[str]`
+            The names of all of the extensions to load.
+        package: :type:`Optional[str]`
+            The package name to resolve relative imports with.
+            This is required when loading an extension using a relative path, e.g ``.foo.test``.
+            Defaults to ``None``.
+        packages: :type:`Optional[List[str]]`
+            A list of packages to resolve relative imports with.
+            This is required when loading an extension using a relative path, e.g ``.foo.test``.
+            Defaults to ``None``.
+        extras: :type:`Optional[List[Dict[str, Any]]]`
+            A list of extra arguments to pass to the extension's setup function.
+
+        Returns
+        --------
+        A :type:`List[str]` that contains the names of all of the extensions
+        that loaded successfully.
+
+        Raises
+        --------
+        ValueError
+            If the ``package`` and ``packages`` parameters are both provided or if
+            the length of ``packages`` or the length of ``extras`` is not equal to 
+            the length of ``names``.
+        ExtensionNotFound
+            An extension could not be imported.
+        ExtensionAlreadyLoaded
+            An extension is already loaded.
+        NoEntryPointError
+            An extension does not have a setup function.
+        ExtensionFailed
+            An extension or its setup function had an execution error.
+        """
+        if package and packages:
+            raise ValueError("Cannot provide both package and packages.")
+
+        if packages:
+            if len(packages) != len(names):
+                raise ValueError("The length of packages must match the length of extensions.")
+            packages_itr = iter(packages)
+
+        if extras:
+            if len(extras) != len(names):
+                raise ValueError("The length of extra parameters must match the length of extensions.")
+            extras_itr = iter(extras)
+
+        loaded_extensions: List[str] = []
+
+        for extension in names:
+            if packages:
+                # this shouldn't give any errors since we already checked the length
+                cur_package = next(packages_itr)
+
+            if extras:
+                # this shouldn't give any errors since we already checked the length
+                cur_extra = next(extras_itr)
+
+            try:
+                if package:
+                    self.load_extension(extension, package=package, extras=cur_extra)
+                elif packages:
+                    self.load_extension(extension, package=cur_package, extras=cur_extra)
+            except Exception:
+                raise
+            else:
+                loaded_extensions.append(extension)
```

Comment:
I could, but it would append the extensions that failed, correct? And I don't want to do that. But, the exception would raise at `load_extension` and stop the code from running, so it does maybe work.